### PR TITLE
SOF-7142: add common cypress dependencies

### PR DESCRIPTION
**Pull request by k0stik**

### CODE REVIEW

File: src/cypress/Browser.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    //                 (error: object, result: { value?: R }) => {
+    //                     if (error) {
+    //                         return reject(error);
+    //                     }
+    //                     resolve(result.value || null);
+    //                 },
+    //             );
+    //         });
+    //     });
+    // }
```
Comment (timurbazhirov) [inline src/cypress/Browser.ts:199]: Let's remove?

File: src/cypress/BrowserFactory.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable class-methods-use-this */
+import { Browser, BrowserSettings } from "./Browser";
+
+export default class BrowserFactory {
```
[the remark this answers is not in the document]
Comment (timurbazhirov) [reply src/cypress/BrowserFactory.ts:4]: we can allow re-defining the Browser in class as a static variable

[the remark this answers is not in the document]
Comment (timurbazhirov) [reply src/cypress/BrowserFactory.ts:4]: Similar to https://github.com/Exabyte-io/mode.js/blob/e7ca0b3a7f8e71a3d3588ebdc602edd6e26aab35/src/methods/factory.js#L4

**Comment by k0stik** [comment]
Moved codebase to TeDe

File: src/cypress/BrowserFactory.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (k0stik) [reply src/cypress/BrowserFactory.ts:4]: Renamed to BrowserManager

File: src/cypress/Browser.ts (hunk 1 above)
Comment (k0stik) [reply src/cypress/Browser.ts:199]: Done

**Pull request closed without merging**